### PR TITLE
CAW-7: Add top-level READMEs

### PR DESCRIPTION
**Pull request by roblafeve**
## Description for [CAW-7](https://crema.atlassian.net/browse/CAW-7)

TLDR:
- Removes default type interfaces
- Adds `./src/components/README.md`, `./src/types/README.md`, and `./src/utils/README.md` that explains intent, usage, and serves as git placeholder ;)

Adds top-level READMEs. This change is based on a conversation we all had about whether or not to include _example_ types. We landed on preferring each top-level directory having a README and keeping the directory visible yet empty that way in addition to giving context and instructions around usage of their respective generators.

#### `./src/components/README.md`

<table>
<tr>
<td>

# `./src/components`

This directory is intended to organize React components.

## Usage

1. In terminal run `npm run new:component` to generate a new `component`
2. Name your component, e.g. `ExampleComponent` (names will be transformed to **Pascal Case**, i.e. `ThisIsPascalCase`)
3. Select whether or not your component will "useState"—selecting "yes" will add a hook.
4. The following directory and files will be generated:
   ```
   src/components/ExampleComponent // component directory
   ├── index.tsx // Define and export the component here
   ├── README.md // Document the component
   ├── stories.tsx // Show the component in various states
   └── test.tsx // Test the component
   ```
5. `ExampleComponent/index.tsx` contains component definition:
   ```tsx
   import React from "react"

   type Props = Readonly<{ name: string }>

   /**
    * ExampleComponent
    * ------------------------------------------------
    * DESCRIPTION_HERE
    */

   export function ExampleComponent({ name }: Props) {
     return <div className="ExampleComponent">{name}</div>
   }
   ```
6. `ExampleComponent/README.md` contains component documentation:
      
        # `<ExampleComponent />`

         DESCRIPTION_HERE

         ## Directory Structure

         - `stories.tsx`: Component playground (`npm run test:playground`)
         - `test.tsx`: Component tests (`npm run test:unit`)
         - `index.tsx`: Component code
         - `README.md`: Component documentation (hey, that's me!)

         ## Example

         ExampleComponent

         > Image location: [`.loki/reference/chrome_ExampleComponent_example.png`](../../../.loki/reference/chrome_ExampleComponent_example.png)
         > 
         > To regenerate: 
         > 1. `npm run test:playground` (skip if running)
         > 1. `npm run test:visual:update -- --storiesFilter="^ExampleComponent example\$"`
7. `ExampleComponent/stories.tsx` contains component stories:
   ```tsx
   import React from "react"
   import decoratorCentered from "@storybook/addon-centered"
   import { ExampleComponent } from "."

   /**
    * See Storybook Docs: Writing Stories
    * https://storybook.js.org/docs/basics/writing-stories/
    */

   export default {
     title: "ExampleComponent",
     decorators: [decoratorCentered],
   }

   export const example = () => <ExampleComponent name="ExampleComponent" />

   ```
8. `ExampleComponent/test.tsx` contains component tests:
   ```tsx      
   import { ExampleComponent } from "."
   import React from "react"
   import { render } from "@testing-library/react"

   describe("ExampleComponent", () => {
     it("name prop is rendered", () => {
       // Arrange
       const name = "ExampleComponent"

       // Act
       const { getByText } = render(<ExampleComponent name={name} />)
       const received = getByText(name)

       // Assert
       expect(received).toBeDefined()
     })
   })
   ```
</td>
</tr>
</table>

#### `./src/types/README.md`

<table>
<tr>
<td>

# `./src/types`

This directory is intended to organize global application [types](http://www.typescriptlang.org/docs/handbook/basic-types.html).

## Usage

1. In terminal run `npm run new:type` to generate a new `type`
2. Name your type, e.g. `ExampleType` (names will be transformed to **Pascal Case**, i.e. `ThisIsPascalCase`)
3. `./src/types/ExampleType.ts` will be generated with the following content: 
   
   ```ts
   /**
    * ExampleType
    * ------------------------------------------------
    * DESCRIPTION_HERE
    */

   export interface ExampleType {}
   ```

4. Add a description to the comment block. This will be important documentation and useful context for yourself whenever you hover over the reference (in VSCode at least).
5. Add some fields to your interface:
   
   ```ts
   /**
    * ExampleType
    * ------------------------------------------------
    * A type intended for example use only 👍
    */

   export interface ExampleType {
     name: string
     age: number
   }
   ```

6. Import elsewhere for use:
   
   ```ts
   import { ExampleType } from "../../types/ExampleType"

   /**
    * exampleUtil
    * ------------------------------------------------
    * A util function intended for example use only 👍
    */

   export const exampleUtil = (x: ExampleType) => x
   ```

</td>
</tr>
</table>

#### `./src/utils/README.md`

<table>
<tr>
<td>

# `./src/utils`

This directory is intended to organize global application utilities—functions, constants, and more.

## FAQ

1. _**Are `utils` different from `helpers`, `lib`, etc.?**_ No. It's jut the name that was chosen.
2. _**What constitutes a `util`?**_ Anything that isn't a React component, type, and doesn't belong to another project-specific grouping of code (e.g. Redux related code, etc.). Application-specific logic and type transformation code are some examples of what would make a good `util`.
3. _**Must `utils` be functions?**_ No. They can be any value you need them to be, but functions tend to be the most common kind of util (hence, we generate on by default).

## Usage

1. In terminal run `npm run new:util` to generate a new `utility`
2. Name your utility, e.g. `exampleUtil` (names will be transformed to **Camel Case**, i.e. `thisIsCamelCase`)
3. The following directory and files will be generated:
   ```
   src/utils
   ├── exampleUtil // Utility directory
   │   ├── index.ts // Define and export the utility here
   │   ├── README.md // Document the utility
   │   └── test.ts // Test the utility
   └── README.md // 👈 you are here!
   ```
4. `exampleUtil/index.ts` contains the utility definition:
   ```ts
   /**
    * exampleUtil
    * ------------------------------------------------
    * DESCRIPTION_HERE
    */

   export const exampleUtil = <A>(x: A) => x
   ```
5. `exampleUtil/README.md` contains the utility documentation:
      
        # `exampleUtil`

        DESCRIPTION_HERE

        ## Example

        ```ts
        import { exampleUtil } from "./exampleUtil"

        exampleUtil(1) // 1
        ```
6. `exampleUtil/test.ts` contains the utility tests:
   ```ts      
   import { exampleUtil } from "."

   describe("exampleUtil", () => {
     it("works", () => {
       // Arrange
       const valA = 1

       // Act
       const received = exampleUtil(valA)
       const expected = 1

       // Assert
       expect(received).toEqual(expected)
     })
   })
   ```

</td>
</tr>
</table>

### 📋 Acceptance Criteria

> Checked off by the PR **Assignee(s)**

- [x] `./src/components`
- [x] `./src/types`
- [x] `./src/utils`

### 👩‍🔬 Test Instructions

1. How to test this PR
2. Prefer bulleted description
3. Start after checking out this branch
4. Include any setup required, such as bundling scripts, restarting services, etc.
5. Include test case, and expected output

## 🔎 Reviewer Checklist

> Checked off by the PR **Reviewers**

### Required

> These always need to be checked

- [ ] Merge destination is correct
- [ ] Code is correct as understood and conforms to quality standards
- [ ] Tests have been added where appropriate (unit, visual, end-to-end)
- [ ] Acceptance Criteria have been met

---

>### Roles & Responsibilities
>
>#### 👨‍💻 Assignee
>
>- Initiator of this PR (be sure to set in GitHub UI)
>- Addresses feedback and change requests
>- Merges PR once approved (usually deletes branch unless `develop` or `release`)
>
>#### 👩‍💻 Reviewer
>
>- Invited to review PR by **Assignee** (via GitHub UI)
>- Is expected to complete a review and address followup

### CODE REVIEW

**Comment by roblafeve** [comment]
@blaketarter @bvarberg I'm curious what your thoughts are on this approach. I've only added the one for `types`, but of course I would include them for each top-level `./src` dir.

**Comment by blaketarter** [comment]
@roblafeve I really like this, it definitely serves as better exposure for the generators that some people might not know about when they use this repo.

**Review by blaketarter — DISMISSED** [review]

**Comment by roblafeve** [comment]
@blaketarter cool, I'll add the same for the others based on this. Any wording that you would change?

**Comment by bvarberg** [comment]
I think it works well! 👍

**Comment by roblafeve** [comment]
All three READMEs have been added. Ready for final review 👍

**Review by Alexandria — APPROVED** [review]
I was able to walk through and test everything with ease! I really like the addition of the FAQ in the`src/utils/README.md`
